# CLPLS_package

## Overview
CLPLS is an efficient spatial deconvolution method that combines graph contrastive learning and partial least squares regression to estimate the cell-type composition at each spatial location. Moreover, CLPLS is a flexible method that it can be extended to integrat spatial transcriptomics data with single-cell multi-omics data, enabling the exploration of the spatial landscape of epigenomics.

## Requirements

* anndata==0.10.5
* networkx==2.8.4
* numpy==1.23.5
* pandas==1.5.3
* scanpy==1.9.8
* scikit_learn==1.2.1
* scipy==1.10.0
* torch==2.3.0
* tqdm==4.64.1

## Data

The preprocessed data used in our experiments have been uploaded and are freely available at https://drive.google.com/drive/folders/12Ma5hciEt454Cohkh2EqUUm49xk9KTpi?usp=sharing.

## Tutorial
For the step-by-step tutorials, we have released them at https://github.com/LindsayMo/CLPLS_package/tree/main/experiment.




## Citation
